### PR TITLE
Tests porting: foundation text and text-field

### PR DESCRIPTION
**Pull request by eymar**

### CODE REVIEW

File: compose/ui/ui-test-junit4/src/skikoMain/kotlin/androidx/compose/ui/test/ComposeUiTest.skikoMain.kt [hunk 1]
```diff
@@ -261,6 +266,16 @@ class SkikoComposeUiTest(
         return surface.makeImageSnapshot()
     }
 
+    fun SemanticsNodeInteraction.captureToImage(): ImageBitmap {
```
Comment (eymar) [inline compose/ui/ui-test-junit4/src/skikoMain/kotlin/androidx/compose/ui/test/ComposeUiTest.skikoMain.kt:269]: API change: new method

**Review by dima-avdeev-jb — APPROVED** [review]

File: compose/ui/ui-test-junit4/src/skikoMain/kotlin/androidx/compose/ui/test/ComposeUiTest.skikoMain.kt (hunk 1 above)
Comment (igordmn) [reply compose/ui/ui-test-junit4/src/skikoMain/kotlin/androidx/compose/ui/test/ComposeUiTest.skikoMain.kt:269]: nit: Can we use `waitForIdle()` + `surface.makeImageSnapshot(iRect)` for better performance?

File: compose/ui/ui-test-junit4/src/skikoMain/kotlin/androidx/compose/ui/test/ComposeUiTest.skikoMain.kt [hunk 2]
```diff
@@ -261,6 +266,16 @@ class SkikoComposeUiTest(
...
+        val rootImage = this@SkikoComposeUiTest.captureToImage()
```
Comment (igordmn) [inline compose/ui/ui-test-junit4/src/skikoMain/kotlin/androidx/compose/ui/test/ComposeUiTest.skikoMain.kt]: `captureToImage` and `SemanticsNodeInteraction.captureToImage()` have different types (Image, ImageBitmap). We should keep only one. We probably should return `ImageBitmap`, so our API will depend on Compose, not Skia.

It is a breaking change, but our test API is still experimental

Comment (eymar) [reply compose/ui/ui-test-junit4/src/skikoMain/kotlin/androidx/compose/ui/test/ComposeUiTest.skikoMain.kt]: I agree. Done.
Also updated some tests which used `fun captureToImage(): Image`

File: compose/ui/ui-test-junit4/src/skikoMain/kotlin/androidx/compose/ui/test/ComposeUiTest.skikoMain.kt [hunk 3]
```diff
@@ -256,9 +261,17 @@ class SkikoComposeUiTest(
         return SemanticsNodeInteractionCollection(testContext, useUnmergedTree, matcher)
     }
 
-    fun captureToImage(): Image {
+    fun captureToImage(): ImageBitmap {
```
Comment (eymar) [inline compose/ui/ui-test-junit4/src/skikoMain/kotlin/androidx/compose/ui/test/ComposeUiTest.skikoMain.kt:264]: Breaking API chnage: Return type was Image, now - ImageBitmap

**Review by igordmn — APPROVED** [review]

**Pull request merged**